pkg/cmd/checkendpoints/controller/connection_checker.go

```diff
@@ -115,29 +112,31 @@ func (c *connectionChecker) Stop() {
 // updateStatus applies updates. If an error occurs applying an update,
 // it remain on the queue and retried on the next call to updateStatus.
 func (c *connectionChecker) updateStatus(ctx context.Context) {
-	c.updatesLock.Lock()
-	defer c.updatesLock.Unlock()
-	if len(c.updates) > 20 {
-		_, _, err := v1alpha1helpers.UpdateStatus(ctx, c.client, c.name, c.updates...)
+	c.updates.Lock()
+	defer c.updates.Unlock()
+	if len(c.updates.Queue()) > 20 {
```

Comment:
make this function return a copy so that you don't need to manage the lock here.

Reply:
> make this function return a copy so that you don't need to manage the lock here.

oh, blech. The `Clear` is being caught up in here too.  

How about just making `UpdateStatus` native on the updatemanager.